File: api/Models/Expense.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    [Table("expenses")]
    public class Expense
    {
        public int Id { get; set; }
        public User? UserId { get; set; }
        public string? Name { get; set; }
        public decimal Amount { get; set; }
        public DateTime Date { get; set; }
    }
}

File: api/internal/models/expense.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Expense struct {
	gorm.Model
	Title  string    `json:"title"`
	Amount float64   `json:"amount"`
	Type   string    `json:"type"`
	Date   time.Time `json:"date"`
	UserID uint      `json:"user_id"`
}
